lib/withOnyx.d.ts

```diff
@@ -0,0 +1,27 @@
+import {IsEqual} from 'type-fest';
+import {DeepKeyOf, TypeOptions} from './types';
+
+type Key = TypeOptions['keys'];
+type Value = TypeOptions['values'];
+
+type Mapping<TComponentProps, TOnyxProps, TMappingKey extends keyof TOnyxProps, TOnyxKey extends Key> = IsEqual<Value[TOnyxKey] | null, TOnyxProps[TMappingKey]> extends true
```

Comment:
@hayata-suenaga @blazejkustra Coming from [here](https://github.com/Expensify/react-native-onyx/pull/267#issuecomment-1632060418).

Here I use `IsEqual` which will require the return type of this `Mapping` to match the return type of the associated prop in `OnyxProps`.

Reply:
`Value[TOnyxKey]` and `TOnyxProps[TMappingKey]` won't be the same if the selector is used to modify the data stored in onyx. please see [this comment](https://github.com/Expensify/react-native-onyx/pull/267#discussion_r1263092472)